tests/pass-dep/shims/libc-misc.rs

```diff
@@ -239,30 +239,34 @@ fn test_localtime_r() {
     };
     let res = unsafe { libc::localtime_r(custom_time_ptr, &mut tm) };
 
-    unsafe { assert_eq!((*tm_ptr).tm_sec, 56) };
-    unsafe { assert_eq!((*tm_ptr).tm_min, 43) };
-    unsafe { assert_eq!((*tm_ptr).tm_hour, 7) };
-    unsafe { assert_eq!((*tm_ptr).tm_mday, 7) };
-    unsafe { assert_eq!((*tm_ptr).tm_mon, 3) };
-    unsafe { assert_eq!((*tm_ptr).tm_year, 124) };
-    unsafe { assert_eq!((*tm_ptr).tm_wday, 0) };
-    unsafe { assert_eq!((*tm_ptr).tm_yday, 97) };
-    unsafe { assert_eq!((*tm_ptr).tm_isdst, -1) };
-    unsafe { assert_eq!((*tm_ptr).tm_gmtoff, 0) };
-    unsafe { assert_eq!(CStr::from_ptr((*tm_ptr).tm_zone).to_str().unwrap(), "+00") };
-
+    assert_eq!(tm.tm_sec, 56);
+    assert_eq!(tm.tm_min, 43);
+    assert_eq!(tm.tm_hour, 7);
+    assert_eq!(tm.tm_mday, 7);
+    assert_eq!(tm.tm_mon, 3);
+    assert_eq!(tm.tm_year, 124);
+    assert_eq!(tm.tm_wday, 0);
+    assert_eq!(tm.tm_yday, 97);
+    assert_eq!(tm.tm_isdst, -1);
+    assert_eq!(tm.tm_gmtoff, 0);
+    unsafe { assert_eq!(CStr::from_ptr(tm.tm_zone).to_str().unwrap(), "+00") };
+
+    let res_tm: libc::tm;
+    unsafe {
+        res_tm = *res;
+    }
     // The returned value should match the pointer passed in.
```

Comment:
Instead of comparing everything a second time, can't you just check that `res` is equal to `&mut tm` (as in, `ptr::eq`)?

Reply:
Yeah, this makes sense.